src/root/async.c

```diff
@@ -109,7 +114,8 @@ unsigned __stdcall startthread(void *p)
     AsyncRead *aw = (AsyncRead *)p;
 
     //printf("aw->filesdim = %p %d\n", aw, aw->filesdim);
-    for (size_t i = 0; i < aw->filesdim; i++)
+    size_t dim = aw->filesdim;
+    for (size_t i = 0; i < dim; i++)
```

Comment:
"... posix [version] `startthread` correctly caches `aw->filesdim`, so it will not use freed memory in loop condition check."

I am not sure if this is needed (I think it should not), but I left it as it is.